TestLinuxCore: two aarch64 reg test cases fail on FreeBSD/aarch64

|  |  |
| --- | --- |
| Bugzilla Link | [49415](https://llvm.org/bz49415) |
| Version | unspecified |
| OS | Linux |
| Reporter | LLVM Bugzilla Contributor |
| CC | @arichardson,@emaste,@JDevlieghere,@labath |

## Extended Description 
FAIL: LLDB (/home/mgorny/llvm-project/build.arm64/bin/clang-aarch64) :: test_aarch64_regs (TestLinuxCore.LinuxCoreTestCase)
FAIL: LLDB (/home/mgorny/llvm-project/build.arm64/bin/clang-aarch64) :: test_aarch64_sve_regs_fpsimd (TestLinuxCore.LinuxCoreTestCase)


The results look a bit suspicious -- like a rounding/printing problem.  I wonder if it could be affected by some subtle stdlib difference.


======================================================================
FAIL: test_aarch64_regs (TestLinuxCore.LinuxCoreTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mgorny/llvm-project/lldb/packages/Python/lldbsuite/test/decorators.py", line 149, in wrapper
    return func(*args, **kwargs)
  File "/home/mgorny/llvm-project/lldb/test/API/functionalities/postmortem/elf-core/TestLinuxCore.py", line 348, in test_aarch64_regs
    substrs=["{} = {}".format(regname, value)])
  File "/home/mgorny/llvm-project/lldb/packages/Python/lldbsuite/test/lldbtest.py", line 2566, in expect
    self.fail(log_msg)
AssertionError: Ran command:
"register read d4"

Got output:
      d4 = 0

Expecting sub string: "d4 = 5.35161536149201e-315" (was not found)
Config=aarch64-/home/mgorny/llvm-project/build.arm64/bin/clang
======================================================================
FAIL: test_aarch64_sve_regs_fpsimd (TestLinuxCore.LinuxCoreTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mgorny/llvm-project/lldb/packages/Python/lldbsuite/test/decorators.py", line 149, in wrapper
    return func(*args, **kwargs)
  File "/home/mgorny/llvm-project/lldb/test/API/functionalities/postmortem/elf-core/TestLinuxCore.py", line 435, in test_aarch64_sve_regs_fpsimd
    substrs=["{} = {}".format(regname, value)])
  File "/home/mgorny/llvm-project/lldb/packages/Python/lldbsuite/test/lldbtest.py", line 2566, in expect
    self.fail(log_msg)
AssertionError: Ran command:
"register read d4"

Got output:
      d4 = 0

Expecting sub string: "d4 = 5.35161536149201e-315" (was not found)
Config=aarch64-/home/mgorny/llvm-project/build.arm64/bin/clang
----------------------------------------------------------------------
Ran 18 tests in 37.985s

RESULT: FAILED (16 passes, 2 failures, 0 errors, 0 skipped, 0 expected failures, 0 unexpected successes)

--- a/lldb/test/API/functionalities/postmortem/elf-core/TestLinuxCore.py
+++ b/lldb/test/API/functionalities/postmortem/elf-core/TestLinuxCore.py
@@ -292,9 +292,7 @@ class LinuxCoreTestCase(TestBase):
         self.dbg.DeleteTarget(target)
 
     @skipIfLLVMTargetMissing("AArch64")
-    @expectedFailureAll(
-        archs=["aarch64"], oslist=["freebsd"], bugnumber="llvm.org/pr49415"
-    )
+    # This test fails on FreeBSD 12 and earlier, see llvm.org/pr49415 for details.
     def test_aarch64_regs(self):
         # check 64 bit ARM core files
         target = self.dbg.CreateTarget(None)
@@ -377,9 +375,7 @@ class LinuxCoreTestCase(TestBase):
         self.expect("register read --all")
 
     @skipIfLLVMTargetMissing("AArch64")
-    @expectedFailureAll(
-        archs=["aarch64"], oslist=["freebsd"], bugnumber="llvm.org/pr49415"
-    )
+    # This test fails on FreeBSD 12 and earlier, see llvm.org/pr49415 for details.
     def test_aarch64_sve_regs_fpsimd(self):
         # check 64 bit ARM core files
         target = self.dbg.CreateTarget(None)

